pnr/openroad/build.sh

```diff
@@ -30,6 +30,14 @@ esac
 # Support `any_cast` workaround: https://conda-forge.org/docs/maintainer/knowledge_base.html#newer-c-features-with-old-sdk
 if [[ $OS == "Mac" ]]; then
     export CXXFLAGS="${CXXFLAGS} -D_LIBCPP_DISABLE_AVAILABILITY"
+
+    # Patch the configuration to use our path, rather than the hardcoded default
+    # of /System/Library/Frameworks/OpenGL.framework/Headers and "/System/Library/Frameworks/AGL.framework/Headers"
+    if grep -q _GL_INCDIRS "$PREFIX/lib/cmake/Qt5Gui/Qt5GuiConfigExtras.cmake"
+    then
+        sed '1,11d' < "$PREFIX/lib/cmake/Qt5Gui/Qt5GuiConfigExtras.cmake" | sed '1 s|.*|set(_qt5gui_OPENGL_INCLUDE_DIR "'$CONDA_BUILD_SYSROOT'/System/Library/Frameworks/OpenGL.framework/Headers")|' > "$PREFIX/lib/cmake/Qt5Gui/Qt5GuiConfigExtras.cmake.bak"
```

Comment:
would this work too?
```
sed -i -e '1,11d' -e  '1 s|.*|set(_qt5gui_OPENGL_INCLUDE_DIR "'$CONDA_BUILD_SYSROOT'/System/Library/Frameworks/OpenGL.framework/Headers")|' $PREFIX/lib/cmake/Qt5Gui/Qt5GuiConfigExtras.cmake
```

Reply:
I think that using a patch would be a safer option though, as the sed command `1,11d` would be super fragile to any change to the underlying `cmake` file.

Reply:
Here is what the patch could look like:
```
--- #<buffer 1>
+++ #<buffer 2>
@@ -1,15 +1,4 @@
-
-
-
-set(_GL_INCDIRS "/opt/conda/conda-bld/qt-main_1660123285207/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/include" "/opt/conda/conda-bld/qt-main_1660123285207/_build_env/x86_64-conda-linux-gnu/sysroot/usr/include" "/opt/conda/conda-bld/qt-main_1660123285207/_build_env/x86_64-conda-linux-gnu/sysroot/usr/include/libdrm" "/opt/conda/conda-bld/qt-main_1660123285207/_build_env/x86_64-conda-linux-gnu/sysroot/usr/include")
-find_path(_qt5gui_OPENGL_INCLUDE_DIR GL/gl.h
-    PATHS ${_GL_INCDIRS}
-)
-if (NOT _qt5gui_OPENGL_INCLUDE_DIR)
-    message(FATAL_ERROR "Failed to find \"GL/gl.h\" in \"${_GL_INCDIRS}\".")
-endif()
-unset(_GL_INCDIRS)
-
+set(_qt5gui_OPENGL_INCLUDE_DIR "'$CONDA_BUILD_SYSROOT'/System/Library/Frameworks/OpenGL.framework/Headers")
 # Don't check for existence of the _qt5gui_OPENGL_INCLUDE_DIR because it is
 # optional.
 

```

Reply:
No, because `sed` on Mac is BSD sed. You could give it an extension like `-i bak` which would use a temporary file named `$FILE.bak`, but that's not portable.

I did it this way so that it would work with both GNU and BSD sed.

Reply:
ah I see, that make sense! do you think the sed line will survive a cosmetic change to the file?

Reply:
I don't think so. At least a patch would explode if things shifted around, e.g. if a new version of qt is released.

Reply:
if you want to inline the patch, we can do something like this:
```
cat <<EOF | patch $PREFIX/lib/cmake/Qt5Gui/Qt5GuiConfigExtras.cmake
--- #<buffer 1>
+++ #<buffer 2>
@@ -1,15 +1,4 @@
-
-
-
-set(_GL_INCDIRS "/opt/conda/conda-bld/qt-main_1660123285207/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/include" "/opt/conda/conda-bld/qt-main_1660123285207/_build_env/x86_64-conda-linux-gnu/sysroot/usr/include" "/opt/conda/conda-bld/qt-main_1660123285207/_build_env/x86_64-conda-linux-gnu/sysroot/usr/include/libdrm" "/opt/conda/conda-bld/qt-main_1660123285207/_build_env/x86_64-conda-linux-gnu/sysroot/usr/include")
-find_path(_qt5gui_OPENGL_INCLUDE_DIR GL/gl.h
-    PATHS \${_GL_INCDIRS}
-)
-if (NOT _qt5gui_OPENGL_INCLUDE_DIR)
-    message(FATAL_ERROR "Failed to find \"GL/gl.h\" in \"\${_GL_INCDIRS}\".")
-endif()
-unset(_GL_INCDIRS)
-
+set(_qt5gui_OPENGL_INCLUDE_DIR "'\$CONDA_BUILD_SYSROOT'/System/Library/Frameworks/OpenGL.framework/Headers")
 # Don't check for existence of the _qt5gui_OPENGL_INCLUDE_DIR because it is
 # optional.
EOF
```

Reply:
That doesn't quite work, but I did come up with a patch that does.

Also, openroad changed and didn't build, so I had to redo the `disable-mpl2-and-par.patch`

Reply:
I went back to the sed approach.

The problem is that the Qt version seems to change, particularly between osx and osx-arm64, so it's difficult to come up with a patch that works for both.